examples/step-55/doc/intro.dox

```diff
@@ -0,0 +1,157 @@
+p4est library installed. The installation of deal.II
+together with these two additional libraries is described in the <a
+href="../../readme.html" target="body">README</a> file. Note also that
+to work properly, this program needs access to the Hypre
+preconditioner package implementing algebraic multigrid; it can be
+installed as part of PETSc but has to be explicitly enabled during
+PETSc configuration; see the page linked to from the installation
+instructions for PETSc.
+
+
+<a name="Intro"></a>
+<h1>Introduction</h1>
+
+Building on step-40, this tutorial shows how to solve linear PDEs with several
+components in parallel using MPI with PETSc or Trilinos for the linear
+algebra. For this, we return to the Stokes equations as discussed in
+step-22. The motivation for writing this tutorial is to provide an
+intermediate step (pun intended) between step-40 (parallel Laplace) and
+step-32 (parallel coupled Stokes with Boussinesq for a time dependent
+problem).
+
+The learning outcomes for this tutorial are:
+
+- You are able how to solve PDEs with several variables in parallel and can
+  apply this to different problems.
+
+- You understand the concept of optimal preconditioners and are able to check
+  this for a particular problem.
+  
+- You are able to construct manufactured solutions using the free computer
+  algreba system SymPy (https://sympy.org).
+
+- You can implement various other tasks for parallel programs: error
+  computation, writing graphical output, etc.
+
+- You can visualize vector fields, stream lines, and contours of vector
+  quantities.
+
+We are solving for a velocity $\textbf{u}$ and pressure $p$ that satisfy the
+Stokes equation, which reads
+@f{eqnarray*}
+  - \triangle \textbf{u} + \nabla p &=& \textbf{f}, \\
+  -\textrm{div}\; \textbf{u} &=& 0.
+@f}
+
+
+<h3>Optimal preconditioners</h3>
+
+Make sure that you read (even better: try) what is described in "Block Schur
+complement preconditioner" in the "Possible Extensions" section in step-22.
+Like described there, we are going to solve the block system using a Krylov
+method and a block preconditioner.
+
+Our goal here is to construct a very simple (maybe the simplest?) optimal
+preconditioner for the linear system. A preconditioner is optimal, if the
+number of iterations of the preconditioned system is independent of the mesh
+size $h$. You can extend that definition to also require indepence of the
+number of processors used (we will discuss that in the results section), the
+computational domain and the mesh quality, the test case itself, the
+polynomial degree of the finite element space, and more.
+
+The preconditioner described here is even simpler than the one described in
+step-22 and will typically require more iterations and consequently time to
+solve. When considering preconditioners, optimality is not the only important
+metric. But an optimal and expensive preconditioner is typically more
+desirable than a cheaper, non-optimal one.
```

Comment:
...because, eventually, as the mesh size becomes smaller and smaller and linear problems become bigger and bigger, the former will beat the latter.